integration/client/client_test.go

```diff
@@ -457,6 +463,28 @@ func TestImagePullWithConcurrencyLimit(t *testing.T) {
 	}
 }
 
+func TestImagePullWithTracing(t *testing.T) {
+	client, err := newClient(t, address)
+	require.NoError(t, err)
+	defer client.Close()
+
+	ctx := namespaces.WithNamespace(context.Background(), "tracing")
+
+	//create in memory exporter and global tracer provider for test
+	exp, tp := newInMemoryExporterTracer()
+	// Shutdown properly so nothing leaks.
+	defer func() { _ = tp.Shutdown(ctx) }()
+
+	//do an image pull which is instrumented, we should expect spans in the exporter
+	_, err = client.Pull(ctx, testImage, WithPlatformMatcher(platforms.Default()))
+	require.NoError(t, err)
+
+	err = tp.ForceFlush(ctx)
+	require.NoError(t, err)
+	assert.NotEmpty(t, exp.GetSpans())
```

Comment:
Besides checking for non Empty, we should also verify that `exp.GetSpans()` actually returns the span that we have created and its status. 

How about adding a validate func that takes in `exp.GetSpans()`, expected spanName (in this case pull.Pull) and expected statusCode and validates that the span in the in memory exporter matches the expected values. 

`exp.GetSpans()` returns SpanStubs which have the necessary info that we can use https://github.com/open-telemetry/opentelemetry-go/blob/bc5cf7eb26a455be6d5b359dea0b6592c4176412/sdk/trace/tracetest/span.go#L28
```
exp.GetSpans()[0].Name()
exp.GetSpans()[0].Status().Code
```

Reply:
Created validateSpan func to address this.